src/atlas/grid/detail/grid/CubedSphere.h

```diff
@@ -208,6 +209,104 @@ class CubedSphere : public Grid {
   // Return number of tiles
   inline idx_t GetNTiles() const { return nTiles_; }
 
+  // Identify tile number from xy space
+  // This is useful when we don't have tile t available
+  inline idx_t tileFromXY( const double xy[] ) const {
+    // Assume one face-edge is of length 90 degrees.
+    //
+    //   y ^
+    //     |
+    //    135              ----------
+    //     |              |     ^    |
+    //     |              |          |
+    //     |              |=<   2   <|
+    //     |              |     v    |
+    //     |              |     =    |
+    //     45  0----------2----------3----------4----------
+    //     |   |    ^     |     ^    |    =     |     =    |
+    //     |   |          |          |    ^     |     ^    |
+    //     |   |=<  0    <|=<   1   <|=<  3    <|=<   4   <|
+    //     |   |    v     |     v    |          |          |
+    //     |   |    =     |     =    |    v     |     v    |
+    //    -45  0 ---------1----------1----------5----------
+    //     |                                    |     =    |
+    //     |                                    |     ^    |
+    //     |                                    |=<   5   <|
+    //     |                                    |          |
+    //     |                                    |     v    |
+    //   -135                                    ----------(5 for end iterator)
+    //     ----0---------90--------180--------270--------360--->  x
+
+    idx_t t{-1};
+
+    if ((xy[LON] >= 0.) && ( xy[LAT] >= -45.) && (xy[LON] < 90.) && (xy[LAT] < 45.)) {
+       t = 0;
+    } else if ((xy[LON] >= 90.) && ( xy[LAT] >= -45.) && (xy[LON] < 180.) && (xy[LAT] < 45.)) {
+       t = 1;
+    } else if ((xy[LON] >= 90.) && ( xy[LAT] >= 45.) && (xy[LON] < 180.) && (xy[LAT] < 135.)) {
+       t = 2;
+    } else if ((xy[LON] >= 180.) && ( xy[LAT] > -45.) && (xy[LON] < 270.) && (xy[LAT] <= 45.)) {
+       t = 3;
+    } else if ((xy[LON] >= 270.) && ( xy[LAT] > -45.) && (xy[LON] < 360.) && (xy[LAT] <= 45.)) {
+       t = 4;
+    } else if ((xy[LON] >= 270.) && ( xy[LAT] > -135.) && (xy[LON] < 360.) && (xy[LAT] <= -45.)) {
+       t = 5;
+    }
+
+    // extra points
+    if ((std::abs(xy[LON]) < 1e-13) && (std::abs(xy[LAT] - 45.) < 1e-13)) t = 0;
+    if ((std::abs(xy[LON] - 180.) < 1e-13) && (std::abs(xy[LAT] + 45.) < 1e-13)) t = 1;
+
+    // for end iterator !!!!
+    if ((std::abs(xy[LON] - 360.) < 1e-13) && (std::abs(xy[LAT] + 135.) < 1e-13)) t = 5;
+
```

Comment:
You should define an epsilon constant somewhere instead of using `double` literals for your tolerance.

Reply:
Done

Reply:
Done!